src/Compilers/CSharp/Portable/Binder/Binder_WithExpression.cs

```diff
@@ -0,0 +1,180 @@
+                        basesBeingResolved: null,
+                        options: LookupOptions.Default,
+                        originalBinder: this,
+                        diagnose: false,
+                        useSiteDiagnostics: ref useSiteDiagnostics);
+                    if (lookupResult.IsSingleViable &&
+                        lookupResult.SingleSymbolOrDefault is var sym)
+                    {
+                        switch (sym.Kind)
+                        {
+                            case SymbolKind.Property:
+                                member = sym;
+                                // PROTOTYPE: this should check for init-only, but that isn't a separate feature yet
+                                if (!(sym is SynthesizedRecordPropertySymbol))
+                                {
+                                    goto default;
+                                }
+                                break;
+
+                            default:
+                                hasErrors = true;
+                                diagnostics.Add(
+                                    ErrorCode.ERR_WithMemberIsNotRecordProperty,
+                                    location);
+                                break;
+                        }
+                    }
+
+                    if (!hasErrors && member is null)
+                    {
+                        hasErrors = true;
+                        Error(
+                            diagnostics,
+                            ErrorCode.ERR_NoSuchMemberOrExtension,
+                            location,
+                            receiverType,
+                            propName);
+                    }
+                }
+
+                var expr = BindValue(initializer.Expression, diagnostics, BindValueKind.RValue);
+                if (!(member is null))
+                {
+                    expr = GenerateConversionForAssignment(
+                        member.GetTypeOrReturnType().Type,
+                        expr,
+                        diagnostics);
+                }
+                lookupResult.Clear();
+                args.Add((member, expr));
+            }
+
+            if (receiverType is null || receiverType.IsVoidType())
+            {
+                diagnostics.Add(ErrorCode.ERR_InvalidWithReceiverType, syntax.Receiver.Location);
+                receiverType = CreateErrorType();
+            }
+
+            if (receiverType.IsErrorType())
+            {
+                lookupResult.Free();
+                return new BoundWithExpression(
+                    syntax,
+                    receiver,
+                    cloneMethod: null,
+                    args.ToImmutableAndFree(),
+                    receiverType,
+                    hasErrors: true);
+            }
+
+            // PROTOTYPE: The receiver type must have a instance method called 'Clone' with no parameters
+            LookupMembersWithoutInheritance(
```

Comment:
>WithoutInheritance [](start = 25, length = 18)

"Without Inheritance" is not justified by the specification.  Should the spec say "The receiver type must **declare** ..." instead of "The receiver type must **contain** ..."?